pkg/apis/provisioning/v1alpha5/kubelet_configuration.go

```diff
@@ -22,4 +27,57 @@ type KubeletConfiguration struct {
+	// eventRecordQPS is the maximum event creations per second. If 0,
+	// there is no limit enforced.
+	EventRecordQPS *int32 `json:"eventRecordQPS,omitempty"`
+	// eventBurst is the maximum size of a burst of event creations,
+	// temporarily allows event creations to burst to this number, while still not exceeding eventRecordQPS.
+	EventBurst *int32 `json:"eventBurst,omitempty"`
+	// registryPullQPS is the limit of registry pulls per second.
+	// The value must not be a negative number. Setting it to 0 means no limit.
+	RegistryPullQPS *int32 `json:"registryPullQPS,omitempty"`
+	// registryBurst is the maximum size of bursty pulls, temporarily allows pulls to burst to this number,
+	// while still not exceeding registryPullQPS.
+	RegistryBurst *int32 `json:"registryBurst,omitempty"`
+	// kubeAPIQPS is the QPS to use while talking with kubernetes apiserver.
+	KubeAPIQPS *int32 `json:"kubeAPIQPS,omitempty"`
+	// kubeAPIBurst is the burst to allow while talking with kubernetes API server.
+	KubeAPIBurst *int32 `json:"kubeAPIBurst,omitempty"`
+	// containerLogMaxSize is a quantity defining the maximum size of the container log file before it is rotated.
+	ContainerLogMaxSize *string `json:"containerLogMaxSize,omitempty"`
+	// containerLogMaxFiles specifies the maximum number of container log files that can be present for a container.
+	ContainerLogMaxFiles *int32 `json:"containerLogMaxFiles,omitempty"`
+	// A comma separated whitelist of unsafe sysctls or sysctl patterns (ending in `∗`).
+	AllowedUnsafeSysctls []string `json:"allowedUnsafeSysctls,omitempty"`
+	// systemReserved is a set of ResourceName=ResourceQuantity (e.g. cpu=200m,memory=150G)
+	// pairs that describe resources reserved for non-kubernetes components.
+	SystemReserved map[string]string `json:"systemReserved,omitempty"`
+	// evictionHard is a map of signal names to quantities that defines hard eviction thresholds.
+	EvictionHard map[string]string `json:"evictionHard,omitempty"`
+}
+
+func (k *KubeletConfiguration) validate() (errs *apis.FieldError) {
+	const notNegative = "cannot be negative"
+	if ptr.Int32Value(k.EventRecordQPS) < 0 {
+		errs = errs.Also(errs, apis.ErrInvalidValue(notNegative, "eventRecordQPS"))
+	}
+	if ptr.Int32Value(k.EventBurst) < 0 {
+		errs = errs.Also(errs, apis.ErrInvalidValue(notNegative, "eventBurst"))
+	}
+	if ptr.Int32Value(k.RegistryPullQPS) < 0 {
+		errs = errs.Also(errs, apis.ErrInvalidValue(notNegative, "registryPullQPS"))
+	}
+	if ptr.Int32Value(k.RegistryBurst) < 0 {
+		errs = errs.Also(errs, apis.ErrInvalidValue(notNegative, "registryBurst"))
+	}
+	if ptr.Int32Value(k.KubeAPIQPS) < 0 {
+		errs = errs.Also(errs, apis.ErrInvalidValue(notNegative, "kubeAPIQPS"))
+	}
+	if ptr.Int32Value(k.KubeAPIBurst) < 0 {
+		errs = errs.Also(errs, apis.ErrInvalidValue(notNegative, "kubeAPIBurst"))
+	}
+	if ptr.Int32Value(k.ContainerLogMaxFiles) < 0 {
+		errs = errs.Also(errs, apis.ErrInvalidValue(notNegative, "containerLogMaxFiles"))
+	}
+	return errs
 }
```

Comment:
It's possible to not have this repetition here and do the same thing in a shorter and more clear way with something like this:

```Go
const notNegative = "cannot be negative"

func (k *KubeletConfiguration) validate() (errs *apis.FieldError) {
	addErrIfNegative(errs, k.EventRecordQPS, "eventRecordQPS")
	addErrIfNegative(errs, k.EventBurst, "eventBurst")
	addErrIfNegative(errs, k.RegistryPullQPS, "registryPullQPS")
	addErrIfNegative(errs, k.RegistryBurst, "registryBurst")
	addErrIfNegative(errs, k.KubeAPIQPS, "kubeAPIQPS")
	addErrIfNegative(errs, k.KubeAPIBurst, "kubeAPIBurst")
	addErrIfNegative(errs, k.ContainerLogMaxFiles, "containerLogMaxFiles")
	return errs
}

func addErrIfNegative(errs *apis.FieldError, num int, name string) {
	if ptr.Int32Value(num) < 0 {
		errs = errs.Also(errs, apis.ErrInvalidValue(notNegative, name))
	}
}
```